crates/fj-core/src/validate/sketch.rs

```diff
@@ -32,13 +32,22 @@ impl SketchValidationError {
         _config: &ValidationConfig,
         errors: &mut Vec<ValidationError>,
     ) {
+        // todo: store referencing objects instead of just a reference count so that we can surface
+        // them in the error message
         let mut referenced_edges =
             std::collections::HashMap::<Handle<HalfEdge>, i32>::new();
+        let mut referenced_cycles =
+            std::collections::HashMap::<Handle<Cycle>, i32>::new();
 
-        // Do we care about how many times each edge is used, or should we just return as soon as
-        // we find one that is used more than once?
         sketch.regions().iter().for_each(|r| {
             r.all_cycles().for_each(|c| {
+                referenced_cycles.insert(c.clone(), {
+                    if let Some(count) = referenced_cycles.get(c) {
+                        count + 1
+                    } else {
+                        1
+                    }
+                });
```

Comment:
I still haven't done a real review (as I said over in https://github.com/hannobraun/fornjot/issues/2133, I'm on vacation right now), but this jumped out to me.

To be clear, this is fine, I would merge it as-is. I just wanted to mention, for the sake of education, that this looks like the perfect use case for the [entry API](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.entry). Something like this should work (not compiled or tested):

```suggestion
                *referenced_cycles.entry(c.clone()).or_insert(0) += 1;
```

Feel free to take this suggestion, or not, as you see fit.